src/compiler/utilitiesPublic.ts

```diff
@@ -403,6 +403,18 @@ namespace ts {
         return !nodeTest || nodeTest(node) ? node : undefined;
     }
 
+    export function getClosestNode<T extends Node>(node: Node, match: (node: Node) => node is T): T;
+    export function getClosestNode<T extends Node>(node: Node, match: (node: Node | undefined) => node is T): T | undefined;
+    export function getClosestNode<T extends Node>(node: Node, match: (node: Node | undefined) => node is T): T | undefined {
+        while (node) {

```

Comment:
It seems the logic duplicates private function `findAncestor` from the compiler utilities
https://github.com/microsoft/TypeScript/blob/97f8b87e34333c40ec6013a327ce65c510df0be0/src/compiler/utilities.ts#L109-L123

Reply:
the util is can public?

Reply:
@a-tarasyuk the util function is can public?

Reply:
@a-tarasyuk